plugins/inputs/logfile/logfile.go

```diff
@@ -35,14 +35,17 @@ type LogFile struct {
 	configs           map[*FileConfig]map[string]*tailerSrc
 	done              chan struct{}
 	removeTailerSrcCh chan *tailerSrc
+	tailerQueue       *tail.TailerEnqueue
 	started           bool
 }
 
 func NewLogFile() *LogFile {
+
 	return &LogFile{
 		configs:           make(map[*FileConfig]map[string]*tailerSrc),
 		done:              make(chan struct{}),
 		removeTailerSrcCh: make(chan *tailerSrc, 100),
+		tailerQueue:       tail.NewTailerFifoQueue(),
```

Comment:
I thought the point was to keep track of all of the monitored log files. Why would we have a separate counter per log file configuration? If I configure to monitor 10 different paths with wildcards in them, this would not communicate across the other routines. It would be misrepresenting the data because if the fd limit is 100 and I have configured 10 different log file monitors, where each path picks up 10 different files, then there are going to be 100 handles opened, but the logging here would only indicate 10 are being used in the log file.

Reply:
Based on my understanding, it's not a separate per log file configuration so the answer is no. In your case, it can be explained by looking through [this code](https://github.com/aws/amazon-cloudwatch-agent/blob/master/plugins/inputs/logfile/logfile.go#L153-L161)
```
for i := range t.FileConfig { // Loop through all in FileConfig
		fileconfig := &t.FileConfig[i]

		targetFiles, err := t.getTargetFiles(fileconfig) // Get all the file if specifies wildcard( e.g log_file_1,log_file_2) 
		if err != nil {
			t.Log.Errorf("Failed to find target files for file config %v, with error: %v", fileconfig.FilePath, err)
		}

		for _, filename := range targetFiles { // Put the tailer
```
However, `tailerQueue:       tail.NewTailerFifoQueue(),` which is in the [init function](https://github.com/aws/amazon-cloudwatch-agent/blob/master/plugins/inputs/logfile/logfile.go#L461-L465) which is created when CloudWatchAgent applies the plugin. Long short story, it is created before `FindLogSrc` and is not created between `FileConfig`

Let me know if that make sense